l2geth/internal/ethapi/api.go

```diff
@@ -893,13 +893,13 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 				}
 				tx := txs[0]
 				blockNumber = tx.L1BlockNumber()
-				timestamp = new(big.Int).SetUint64(tx.L1Timestamp())
+				timestamp = tx.L1Timestamp()
 			}
 		}
 	}
 
 	// Create new call message
-	msg := types.NewMessage(addr, args.To, 0, value, gas, gasPrice, data, false, &addr, nil, types.QueueOriginSequencer)
+	msg := types.NewMessage(addr, args.To, 0, value, gas, gasPrice, data, false, &addr, blockNumber, timestamp, types.QueueOriginSequencer)
```

Comment:
The missing `blockNumber` looks like a bug?

Reply:
I think it was a bug if you were not using the OVM here (because we had the UsingOVM block below before that would set the blocknum and timestamp)